### PR TITLE
Cisco Spark collaboration and integration pack

### PR DESCRIPTION
**Pull request by tonybaloney**
### Status 

- new pack, complete

### Description

This pack integrates into Cisco Spark's public API and supports private and beta instances, it can be used to send messages, create teams, rooms and configure webhooks.

### Checklist (tick everything that applies)

- [x] Metadata: pack.yaml, icon, structure (required for new packs)
- [ ] Version bump (required for changed packs)
- [x] Code linting (required, can be done after the PR checks)
- [ ] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)

### CODE REVIEW

**Comment by tonybaloney** [comment]
having some unicode fun.

**Comment by tonybaloney** [comment]
@Kami or @emedvedev this is done now, just finished testing.

**Comment by Kami** [comment]
Nice work!

Will have a look shortly.

File: packs/ciscospark/actions/create_webhook.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    required: true
+    description: "The event type for the webhook (see doc)"
+  filter:
+    type: string
+    required: false
+    description: "Filter event matching"
+  secret:
+    type: string
+    required: false
+    description: "Secret used to generate paylaod signature"
```
Comment (Kami) [inline packs/ciscospark/actions/create_webhook.yaml]: paylaod -> payload

File: packs/ciscospark/actions/generic_action.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class GenericAction(CiscoSparkAction):
+
+    def run(self, accessor, method_name, **kwargs):
+        a = getattr(self.connection, accessor)
+        result = getattr(a, method_name)(**kwargs)
+        if method_name == 'list':
+            result = list(result)  # iterate generator
+        if isinstance(result, list):
+            return [x._json for x in result]
```
Comment (Kami) [inline packs/ciscospark/actions/generic_action.py]: Does spark library utilize requests underneath or?

**Comment by Kami** [comment]
Merged, thanks!

I will fix the typo locally and also test it.

**Comment by Kami** [comment]
For consistency with other packs are you ok with me renaming it to `cisco_spark`?

**Pull request merged**